p2p/transport/webrtc/stream_write.go

```diff
@@ -193,9 +151,12 @@ func (s *stream) CloseWrite() error {
 		return nil
 	}
 	s.sendState = sendStateDataSent
-	if err := s.sendControlMessage(&pb.Message{Flag: pb.Message_FIN.Enum()}); err != nil {
+	select {
+	case s.sendStateChanged <- struct{}{}:
```

Comment:
The channel will keep the message, no? when the Write goroutine comes around to select block? 
There's a problem with concurrent Write calls but that's a race anyway

Reply:
I overlooked the fact that it is buffered.
It looks correct then, I would rather use already existing primitives than reimplementing what is well tested in the `sync` package but if you think this is fine then keep it.

Reply:
I don't mind changing it if the performance is better, but since this PR didn't introduce this particular structure, let's take this in a separate PR.

Reply:
Using a channel as a "is this changed" flag is a common idiom, assuming there's only _one_ reader. `sync.Cond` is mostly useful when there are many readers.